torch/csrc/lazy/core/shape.h

```diff
@@ -17,24 +18,60 @@ class TORCH_API Shape {
 
   std::string to_string() const;
 
-  c10::ScalarType scalar_type() const { return scalar_type_; }
-  void set_scalar_type(at::ScalarType value) { scalar_type_ = value; }
+  c10::ScalarType scalar_type() const {
+    return scalar_type_;
+  }
+  void set_scalar_type(at::ScalarType value) {
+    scalar_type_ = value;
+  }
+
+  int64_t dim() const {
+    return sizes_.size();
+  }
+  c10::ArrayRef<int64_t> sizes() const {
+    return sizes_;
+  }
+  int64_t size(int64_t dim) const {
+    return sizes_.at(dim);
+  }
+  void set_size(int64_t dim, int64_t size) {
+    sizes_.at(dim) = size;
+  }
+
+  const at::optional<std::vector<bool>>& is_symbolic() const {
+    return is_symbolic_;
+  }
+  c10::SymbolicShape get_symbolic_shape() const;
+
+  // Only sets is_symbolic, doesn't set concrete sizes
+  // Also throws away any information about which symbolic dims
+  // are the same size
+  void set_from_symbolic(c10::SymbolicShape&);
```

Comment:
I'm not sure this is safe API. Imagine, some node already consumed a `Shape` and computed a hash. If we call `set_for_symbolic` on that shape, the hash would be wrong? 

Should we just return a copy with the modified properties @wconstab

Reply:
I've made a new function to return a new copy